### PR TITLE
Use git submodule

### PR DESCRIPTION
**Pull request by tylerhawkes**
This is a fix for https://github.com/boncheolgu/tflite-rs/issues/6. Because of the way tflite downloads dependencies, the easiest thing to do was to run the download script and commit it to the repository, so that accounts for 99% of lines changed, even though I removed most of the downloaded files like docs and tests. Most of the changes are in Cargo.toml where I removed all of the downloading dependencies. I also removed the feature for docs since this should work now. A build feature was added so that a precompiled libtensorflow-lite can be used without spending all of the time compiling.

The build.rs was changed a lot, so I ran `rustfmt` on it. The only command left is the make command and all the other sed commands have been changed to be more portable and hopefully more readable. 

I know this is a large change, and I built it with and without the build and debug features to make sure it works. I'm also happy to move the file structure around if you want it changed.

### CODE REVIEW

**Comment by tylerhawkes** [comment]
Maybe we should create a new repo that has the downloads and then make that a submodule. What do you think?

**Comment by boncheolgu** [comment]
> Maybe we should create a new repo that has the downloads and then make that a submodule. What do you think?

I think that it's a good idea to create a new dedicated repo.

File: build.rs [hunk 1]
```diff
@@ -1,168 +1,151 @@
...
+                }
+            }
+
+            make.arg(format!("TARGET={}", target))
+                .arg(format!("TARGET_ARCH={}", arch))
+                .arg("-j")
+                // allow parallelism to be overridden
+                .arg(
+                    env::var("TFLITE_RS_MAKE_PARALLELISM")
+                        .unwrap_or(env::var("NUM_JOBS").unwrap_or_else(|_| "4".to_string())),
```
[the remark this answers is not in the document]
Comment (tylerhawkes) [reply build.rs]: It doesn't seem like depending on num_cpus is worth it. As long as you're using cargo you get NUM_JOBS always set and you can override it with TFLITE_RS_MAKE_PARALLELISM if you're not using cargo. It's just a last resort default without having to depend on num_cpus.

File: build.rs [hunk 2]
```diff
@@ -1,168 +1,151 @@
...
-        );
-
-        download(&tf_src_url, &tf_src_name).unwrap();
-
-        assert!(
-            is_valid_tf_src(&tf_src_name),
-            "{} is not valid",
-            tf_src_name.to_str().unwrap()
-        );
+    println!("Building tflite");
```
[the remark this answers is not in the document]
Comment (tylerhawkes) [reply build.rs]: You only get these messages if you pass some level of `-v` to Cargo. I was using `-vvv` and could see them then. They can always be found at `$OUT_DIR/output`

File: build.rs [hunk 3]
```diff
@@ -198,10 +181,10 @@ fn import_tflite_types<P: AsRef<Path>>(tflite: P) {
         .derive_partialeq(true)
         .derive_eq(true)
         .header("csrc/tflite_wrapper.hpp")
-        .clang_arg(format!("-I{}", tflite.as_ref().to_str().unwrap()))
+        .clang_arg(format!("-I{}/tensorflow", submodules_str))
         .clang_arg(format!(
-            "-I{}/tensorflow/lite/tools/make/downloads/flatbuffers/include",
-            tflite.as_ref().to_str().unwrap()
+            "-I{}/downloads/flatbuffers/include",
+            submodules_str
```
[the remark this answers is not in the document]
Comment (tylerhawkes) [reply build.rs]: So we need that for bindgen but not for compilation? If so, I'll figure out what to do.

[the remark this answers is not in the document]
Comment (boncheolgu) [reply build.rs]: Yes, polymorphic `NativeTable` is required for `import_tflite_types` and `build_inline_cpp`, not for `libtensorflow-lite.a`.

**Comment by tylerhawkes** [comment]
Now that I have the tests passing I'm going to make a git repo to hold the results of the downloads for merging and then you can take it over before or after we merge. I also need to rebase or squash all the commits so that we don't have that history in the git repo.

**Comment by boncheolgu** [comment]
This looks great!
Could you fix build failure under a feature `generate_model_apis` enabled?

File: build.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (tylerhawkes) [reply build.rs]: So I committed the change to the downloads repo and when building it replaces the source back to the original.

**Comment by tylerhawkes** [comment]
I fixed the issue with the `generate_model_apis` feature. I wasn't using `std::io::Write`. It looks like the builds are all passing. Will you please squash this on merge so we don't have 600k lines of additions and deletions in the history?

**Pull request merged**